descriptors/descriptors_test.go

```diff
@@ -1,114 +1,285 @@
 package descriptors
 
 import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 )
 
-func TestDescriptorSingle(t *testing.T) {
-	descriptorStr := "wpkh(xpub6BzikmgQmvoYG3ShFhXU1LFKaUeU832dHoYL6ka9JpCqKXr7PTHQHaoSMbGU36CZNcoryVPsFBjt9aYyCQHtYi6BQTo6VfRv9xVRuSNNteB/*)"
-	descriptor, err := NewDescriptor(descriptorStr)
-	require.NoError(t, err)
-	defer descriptor.Close()
-	require.Equal(t, 1, descriptor.MultipathLen())
-
-	addr, err := descriptor.AddressAt(NetworkMainnet, 0, 0)
-	require.NoError(t, err)
-	require.Equal(t, "bc1qaz3jjsgpe29v5yzrvd58hsjgx5a9msujsgmyte", addr)
+const (
+	testdataDir           = "testdata"
+	derivationTestVectors = "derivation.json"
+)
 
-	_, err = descriptor.AddressAt(NetworkMainnet, 1, 0)
-	require.Error(t, err)
+func parseNetwork(t *testing.T, network string) Network {
+	switch network {
+	case "mainnet":
+		return NetworkMainnet
+	case "testnet":
+		return NetworkTestnet
+	case "regtest":
+		return NetworkRegtest
+	default:
+		require.FailNow(t, "unknown network: %s", network)
+		return 0
+	}
 }
 
-func TestDescriptorMultipath(t *testing.T) {
-	descriptorStr := "wpkh(xpub6BzikmgQmvoYG3ShFhXU1LFKaUeU832dHoYL6ka9JpCqKXr7PTHQHaoSMbGU36CZNcoryVPsFBjt9aYyCQHtYi6BQTo6VfRv9xVRuSNNteB/<0;1>/*)"
-	descriptor, err := NewDescriptor(descriptorStr)
+func TestDerivationVectors(t *testing.T) {
+	vectors, err := readDerivationTestVectors()
 	require.NoError(t, err)
-	defer descriptor.Close()
 
-	require.Equal(t, 2, descriptor.MultipathLen())
+	for _, vector := range vectors {
+		t.Run(vector.Name, func(tt *testing.T) {
```

Comment:
Is `tt` common practice? It seems safer to be me shadow the other `t` variable, so it's not possible to call the wrong one by accident.